### PR TITLE
fix: throw an error when using an encrypted s3 bucket for capturing alb logs

### LINKED ISSUE #21947: elbv2: cannot use load balancer access logs when bucket is encrypted with KMS key
### Describe the bug

I am running into almost exactly what is described in [this previous issue](https://github.com/aws/aws-cdk/issues/8113) and similar to [this StackOverflow post](https://stackoverflow.com/questions/72009403/how-do-i-enable-access-log-for-load-balancer-using-applicationloadbalancedfargat) where I have:

```ts
albFargateService.loadBalancer.logAccessLogs(bucket, 'alb-access')
```

And am receiving 

```
8:58:34 AM | UPDATE_FAILED        | AWS::ElasticLoadBalancingV2::LoadBalancer     | BotAlbFargateServiceLB34
7310A1
Access Denied for bucket: <my-fake-bucket>. Please check S3bucket permission (Service: AmazonElasticL
oadBalancing; Status Code: 400; Error Code: InvalidConfigurationRequest; Request ID: e95cece4-c4af-4b8f-a6f1
-d1eb0d2e2778; Proxy: null)
```

Am I missing an additional piece of configuration for the bucket to enable access logs?

### Expected Behavior

Access logs are enabled and emitted to bucket

### Current Behavior

Running `cdk deploy` with the snippet shown in the section above we are presented with what appears to be the correct permissions

```
IAM Statement Changes
┌───┬──────────────────────┬────────┬──────────────────────┬──────────────────────┬────────────────────────┐
│   │ Resource             │ Effect │ Action               │ Principal            │ Condition              │
├───┼──────────────────────┼────────┼──────────────────────┼──────────────────────┼────────────────────────┤
│ + │ ${Bucket.Arn}        │ Allow  │ s3:GetBucketAcl      │ Service:delivery.log │                        │
│   │                      │        │                      │ s.amazonaws.com      │                        │
├───┼──────────────────────┼────────┼──────────────────────┼──────────────────────┼────────────────────────┤
│ + │ ${Bucket.Arn}/elb-ac │ Allow  │ s3:Abort*            │ AWS:arn:${AWS::Parti │                        │
│   │ cess/AWSLogs/1234567 │        │ s3:PutObject         │ tion}:iam::123456789 │                        │
│   │ 53096/*              │        │ s3:PutObjectLegalHol │ 021:root             │                        │
│   │                      │        │ d                    │                      │                        │
│   │                      │        │ s3:PutObjectRetentio │                      │                        │
│   │                      │        │ n                    │                      │                        │
│   │                      │        │ s3:PutObjectTagging  │                      │                        │
│   │                      │        │ s3:PutObjectVersionT │                      │                        │
│   │                      │        │ agging               │                      │                        │
│ + │ ${Bucket.Arn}/elb-ac │ Allow  │ s3:PutObject         │ Service:delivery.log │ "StringEquals": {      │
│   │ cess/AWSLogs/1234567 │        │                      │ s.amazonaws.com      │   "s3:x-amz-acl": "buc │
│   │ 53096/*              │        │                      │                      │ ket-owner-full-control │
│   │                      │        │                      │                      │ "                      │
│   │                      │        │                      │                      │ }                      │
├───┼──────────────────────┼────────┼──────────────────────┼──────────────────────┼────────────────────────┤
│ + │ ${Bucket/Key.Arn}    │ Allow  │ kms:Decrypt          │ AWS:arn:${AWS::Parti │                        │
│   │                      │        │ kms:Encrypt          │ tion}:iam::123456789 │                        │
│   │                      │        │ kms:GenerateDataKey* │ 021:root             │                        │
│   │                      │        │ kms:ReEncrypt*       │                      │                        │
└───┴──────────────────────┴────────┴──────────────────────┴──────────────────────┴────────────────────────┘

```

And we are receiving the following as CDK starts to deploy the changes

```
8:58:34 AM | UPDATE_FAILED        | AWS::ElasticLoadBalancingV2::LoadBalancer     | BotAlbFargateServiceLB34
7310A1
Access Denied for bucket: <my-fake-bucket>. Please check S3bucket permission (Service: AmazonElasticL
oadBalancing; Status Code: 400; Error Code: InvalidConfigurationRequest; Request ID: e95cece4-c4af-4b8f-a6f1
-d1eb0d2e2778; Proxy: null)
```

### Reproduction Steps

https://github.com/josefaidt/cdk-accesslogs-repro

Our repository and code can be found here https://github.com/aws-amplify/discord-bot/blob/main/cdk/src/components/hey-amplify-app.ts#L250

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

2.39.1 (build f188fac)

### Framework Version

2.39.1

### Node.js Version

v18.7.0

### OS

macos

### Language

Typescript

### Language Version

4.8.2

### Other information

_No response_

### PR DESCRIPTION
**Pull request by devnotfound**
Currently if one tries to capture alb logs into an encrypted S3 bucket that uses a KMS key, they get an `Access Denied for bucket` error. This association is currently unsupported, however it delays the error at the time of the bucket association. 

This PR now introduces an upfront check that would error saying `Encryption key detected. Bucket encryption using KMS keys is unsupported` before the alb to s3 bucket log association is established. 

This closes #21947 

----

### All Submissions:

* [✔️] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [✖️ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ✖️ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-elasticloadbalancingv2/README.md [hunk 1]
```diff
@@ -77,6 +77,8 @@ const securityGroup2 = new ec2.SecurityGroup(this, 'SecurityGroup2', { vpc });
 lb.addSecurityGroup(securityGroup2);
 ```
 
+NOTE: Enabling ALB logs using an encryption enabled Amazon S3 Bucket is currently unsupported
```
Comment (corymhall) [inline packages/@aws-cdk/aws-elasticloadbalancingv2/README.md:80]: Are you sure this is the case? The [docs](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/load-balancer-access-logs.html#access-logging-bucket-requirements) state that

> With Network Load Balancer access logs, you can't use AWS managed keys, you must use customer managed keys.

**Review by corymhall — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-elasticloadbalancingv2/README.md (hunk 1 above)
Comment (peterwoodworth) [reply packages/@aws-cdk/aws-elasticloadbalancingv2/README.md:80]: In the linked issue there's a reproduction repo provided that uses a bucket with encryption set to KMS - toggling this setting on and off will allow deployment to either fail or succeed. Unless I'm mistaken this setting is creating a CMK

Comment (devnotfound) [reply packages/@aws-cdk/aws-elasticloadbalancingv2/README.md:80]: So, this is what the ALB doc says
```Create an S3 bucket
When you enable access logs, you must specify an S3 bucket for the access logs. The bucket must meet the following requirements.

Requirements

The bucket must be located in the same Region as the load balancer. The bucket and the load balancer can be owned by different accounts.

The prefix that you specify must not include AWSLogs. We add the portion of the file name starting with AWSLogs after the bucket name and prefix that you specify.

The only server-side encryption option that's supported is Amazon S3-managed keys (SSE-S3). For more information, see [Amazon S3-managed encryption keys (SSE-S3)](https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingServerSideEncryption.html).

I can change the README to say - 
`NOTE: Enabling ALB logs using an encryption enabled Amazon S3 Bucket is only supported when using Amazon S3-managed keys (SSE-S3)`
Would that help?

Comment (corymhall) [reply packages/@aws-cdk/aws-elasticloadbalancingv2/README.md:80]: I believe the `BaseLoadBalancer` class is used by _both_ the `ApplicationLoadBalancer` and the `NetworkLoadBalancer`. If there is a difference between behavior then we need to handle it in the specific subclass.

Comment (devnotfound) [reply packages/@aws-cdk/aws-elasticloadbalancingv2/README.md:80]: Yea, my bad, should have checked the inheritance tree. I can certainly move this change to the `ApplicationLoadBalancer` class and rejig the tests

**Comment by TheRealAmazonKendra** [comment]
@Mergifyio update

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
In addition to the previous feedback, please make sure that your PR title confirms to the conventional commit standard (fix, feat, chore) and that it is written in a style that will reflect correctly in the change log (See [Contributing Guide, Pull Requests](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md#step-4-pull-request)).

**Pull request closed without merging**